catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/workspace/transformer/impl/QueryTransformer.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.metacard.workspace.transformer.impl;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.codice.ddf.catalog.ui.metacard.workspace.QueryMetacardImpl;
+import org.codice.ddf.catalog.ui.query.transform.QueryBasicValueTransformer;
+import org.codice.ddf.catalog.ui.query.transform.QuerySortsValueTransformer;
+import org.codice.ddf.catalog.ui.query.transform.QueryValueTransformer;
+import org.codice.ddf.catalog.ui.util.EndpointUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class QueryTransformer {
+
+  private EndpointUtil util;
+
+  private List<QueryValueTransformer> queryValueTransformers;
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(QueryTransformer.class);
+
+  public QueryTransformer(EndpointUtil util) {
+    this.util = util;
+
+    this.queryValueTransformers =
+        Arrays.asList(new QuerySortsValueTransformer(), new QueryBasicValueTransformer());
```

Comment:
The order of these transformers matters, right? The "sorts" attribute could match both if its value is `Serializable`.

Reply:
Yes it does. I've added a comment about this for now but we could come up with a better way of doing this. I could construct a queue every time a query needs to be transformed which would more explicitly describe this process.